### DISCUSSION TITLE
MBR partition table on older installs?

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#7007] [Unsupported Versions] [answered]
I've got a distributed install, done from the SO ISO. Two search nodes were originally installed with 2.3.5x last May, and a couple of months ago I installed another two search nodes, I think with 2.3.7x. (I'm currently on 2.9.x)

The two original search nodes have MBR (DOS) partition tables.  The newer ones have GPT.

Was there a change in how the ISO installer did the disk partitioning since 2.3.5x?  I'd just like to understand why the partition tables are different?

**Comment by dougburks** [comment, marked as answer]
> Was there a change in how the ISO installer did the disk partitioning since 2.3.5x?

I don't believe we've made any major changes to that since 2.3.10:
https://docs.securityonion.net/en/2.3/release-notes.html

> I'd just like to understand why the partition tables are different?

Perhaps some of your machines are configured for BIOS and others are configured for UEFI.

**Reply by abesinger** [reply]
I've solved it, and it wasn't BIOS vs UEFI, though that was a good suggestion. Both the older and newer systems are configured for BIOS. In short, the reason for the difference is that original systems had 1Tb disks, and the newer ones have 4Tb disks. From that, I made a guess that the Centos installer (Anaconda) uses MBR on disks < 2Tb, and I just confirmed that [here](https://docs.centos.org/en-US/centos/install-guide/StorageSpoke-x86/#sect-bootloader-mbr-gpt-x86).

> If the disk is not formatted, or the user erased all partitions from the disk, Anaconda will use:
> 
>- MBR if the disk has less than 232 sectors. Most commonly, disks sectors are 512 bytes in size, in which case this would be equivalent to 2 TiB.
>
>- GPT if the disk has 232 sectors or more.

So [I learned something today.](https://www.youtube.com/watch?v=W3sX3aD6GSM)

The reason that I stumbled across the problem is that my SO systems are VMs (VMWare), and I was trying to increase capacity by enlarging the virtual disk instead of adding another disk. It's a rather arcane problem, andI don't think it would ever have come up otherwise; if I had a physical system I wouldn't be magically enlarging the existing disk.   So it's not going to affect many people, but I figure it's worth documenting for posterity.

I could have taken other paths (where I was headed when I posted #6945), such as just adding disk and using LVM, but I was trying to be clever and keep all of my search nodes looking the same.  And it would have worked if it were for those pesky kids, I mean, that pesky MBR. 

FWIW, It _is_ possible to force the installer to use GPT everywhere (I don't know if that's a desirable default change for the SO ISO) but it can be done from the boot command line. From the [same page referenced above](https://docs.centos.org/en-US/centos/install-guide/StorageSpoke-x86/#sect-bootloader-mbr-gpt-x86):

> Append the inst.gpt option to the boot command line to override the default behavior and use GPT on a disk of less than 232 sectors in size. 

To solve my problem (adding more disk space), I'm going back to the beginning: resizing the disk back down to 1Tb, and then adding a second 3Tb disk and  using LVM to grow the logical volumes. I considered migrating the whole thing to a new 4Tb disk, but I don't want to spend the time fiddling with GRUB to get it to boot. Another option would be to install a new search node and just transfer the indices over from a snapshot, but that too has its own complications.
